moabb/evaluations/base.py

```diff
@@ -216,26 +209,43 @@ def process(self, pipelines, param_grid=None, postprocess_pipeline=None):
             if not (isinstance(pipeline, BaseEstimator)):
                 raise (ValueError("pipelines must only contains Pipelines " "instance"))
 
-        res_per_db = []
-        for dataset in self.datasets:
-            log.info("Processing dataset: {}".format(dataset.code))
-            process_pipeline = self.paradigm.make_process_pipelines(
+        # Prepare dataset processing parameters
+        processing_params = [
+            (
                 dataset,
-                return_epochs=self.return_epochs,
-                return_raws=self.return_raws,
-                postprocess_pipeline=postprocess_pipeline,
-            )[0]
-            # (we only keep the pipeline for the first frequency band, better ideas?)
-
-            results = self.evaluate(
+                self.paradigm.make_process_pipelines(
+                    dataset,
+                    return_epochs=self.return_epochs,
+                    return_raws=self.return_raws,
+                    postprocess_pipeline=postprocess_pipeline,
+                )[0],
+            )
+            for dataset in self.datasets
+        ]
+
+        # Parallel processing...
+        parallel_results = Parallel(
+            n_jobs=self.n_jobs,
+            return_as="generator",
+        )(
+            delayed(self.evaluate)(
                 dataset,
                 pipelines,
                 param_grid=param_grid,
                 process_pipeline=process_pipeline,
                 postprocess_pipeline=postprocess_pipeline,
             )
+            for dataset, process_pipeline in processing_params
+        )
+
+        res_per_db = []
+        # Process results in order
+        for (dataset, process_pipeline), results in zip(
+            processing_params, parallel_results
+        ):
             for res in results:
                 self.push_result(res, pipelines, process_pipeline)
```

Comment:
For a future PR: we should push the results as soon as we obtain them (within the evaluate function), in case the evaluation fails at some point. Ohterwise we can lose all computations